php/WP_Mock/Functions/Handler.php

```diff
@@ -0,0 +1,129 @@
+ */
+class Handler
+{
+    /**
+     * Mocked method handlers registered by the test class.
+     *
+     * @var callable[]|callable-string[]
+     */
+    private static array $handlers = [];
+
+    /**
+     * Overrides any existing handlers to set a new callback.
+     *
+     * @param string|callable-string $function function name
+     * @param array<string|object|Mock, callable-string|string>|callable $callback
+     * @return void
+     */
+    public static function registerHandler(string $function, $callback): void
+    {
+        /** @phpstan-ignore-next-line */
+        self::$handlers[$function] = $callback;
+    }
+
+    /**
+     * Handles a mocked function call.
+     *
+     * @param string|callable-string $functionName function name
+     * @param array<mixed> $args function arguments
+     * @return mixed
+     * @throws ExpectationFailedException
+     */
+    public static function handleFunction(string $functionName, array $args = [])
+    {
+        if (self::handlerExists($functionName)) {
+            $callback = self::$handlers[$functionName];
+
+            return call_user_func_array($callback, $args);
+        } elseif (WP_Mock::strictMode()) {
+            throw new ExpectationFailedException(sprintf('No handler found for %s', $functionName));
+        }
+
+        return null;
+    }
+
+    /**
+     * Checks if a handler exists.
+     *
+     * @param string|callable-string $functionName
+     * @return bool
+     */
+    public static function handlerExists(string $functionName): bool
+    {
+        return isset(self::$handlers[$functionName]);
+    }
+
+    /**
+     * Clears all registered handlers.
+     *
+     * @return void
+     */
+    public static function cleanup(): void
+    {
+        self::$handlers = [];
+    }
+
+    /**
+     * Helper function for common passthru return functions.
+     *
+     * @param string $functionName function name
+     * @param array<mixed> $args function args
+     * @return mixed
+     * @throws ExpectationFailedException
+     */
+    public static function handlePredefinedReturnFunction(string $functionName, array $args = [])
+    {
+        $result = self::handleFunction($functionName, $args);
+
+        if (! self::handlerExists($functionName)) {
+            $result = $args[0] ?? $result;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Helper function for common echo functions.
+     *
+     * @param string $functionName function name
+     * @param array<mixed> $args function arguments
+     * @return void
+     * @throws Exception|ExpectationFailedException
+     */
+    public static function handlePredefinedEchoFunction(string $functionName, array $args = []): void
+    {
+        ob_start();
+
+        try {
+            self::handleFunction($functionName, $args);
+        } catch (Exception $exception) {
+            ob_end_clean();
+
+            throw $exception;
+        }
+
+        $result = ob_get_clean() ?: '';
```

Comment:
Do you expect the buttfer to be closed at anytime while handling the function? I see the catched expception get thrown right after ending/closing the buffer

Reply:
the exception is caught but then thrown again before closing the buffer output

when not thrown, the buffer is captured

Reply:
So we don't need the `?: ''` in this case right?

```suggestion
        $result = ob_get_clean();
```

Reply:
https://github.com/10up/wp_mock/pull/219/commits/71c133d45997f4522f048881c29fc5f7d30f2082 we can, and maybe it's a good idea to throw an exception if `ob_get_clean` returns false (not a string)